Add rule to check if a `*hrl` file is allowed to be included in a module or not (#256)
inaka/elvis_core: doc_rules/elvis_style/no_hrl_include.md

> # Prevent including *.hrl file

Reviewer: ```suggestion
# No hrl include
```

> 

Reviewer: We'll need to decide on a "since" if we publish this.

> With this rule we can raise warning/errors if developer includes restricted `*.hrl` files in non-allowed modules

Reviewer: ```suggestion
This rule raises a warning if specific `hrl` files are included in chosen modules.
```

> 

Reviewer: Use a single empty line here.

```erlang
-module(user).
-include("private.hrl").
.............
```

> It will raise style warning/error for above module 

Reviewer: ```suggestion
E.g. the following will raise a warning for `private.hrl` being included by module `user`.
```

> 

Reviewer: We'll have to add `Works on .beam file? ...!` if we publish this.

## Options

- None.

## Example

```erlang

> {elvis_style, no_hrl_include, #{restricted_hrl_files => ["private.hrl"]}}

Reviewer: ```suggestion
{elvis_style, no_hrl_include, #{ restricted_hrl_files => ["private.hrl"] }}
```

# Prevent including *.hrl file 

The idea behind this rule comes from [this email](http://erlang.org/pipermail/erlang-questions/2017-April/092112.html) by @kvakvs.

> Basically, the warning is emitted if a developer _forgets a comma_ and writes something like the following:

Reviewer 2: I don't think this text matches your ideas, @amarBitMan

Author: yes. I was trying to use the same `.md `format which is used in this project. forgot to remove copied text. I have removed it now.
